### ISSUE TITLE
Question About the ConditionalBlock and LayerNorm

### ISSUE THREAD

**Issue by Adamscc** [lucas-maes/le-wm#46]
Hi, I was trying to reproduce your model architecture until I found that you used **extra** LayerNorm in Attention and FeedForward:
`        self.net = nn.Sequential(
            **nn.LayerNorm(dim),**
            nn.Linear(dim, hidden_dim),
            nn.GELU(),
            nn.Dropout(dropout),
            nn.Linear(hidden_dim, dim),
            nn.Dropout(dropout),
        )
`
`        **x = self.norm(x)**
        drop = self.dropout if self.training else 0.0
        qkv = self.to_qkv(x).chunk(3, dim=-1)  # q, k, v: (B, heads, T, dim_head)
        q, k, v = (rearrange(t, "b t (h d) -> b h t d", h=self.heads) for t in qkv)
        out = F.scaled_dot_product_attention(q, k, v, dropout_p=drop, is_causal=causal)
        out = rearrange(out, "b h t d -> b t (h d)")
        return self.to_out(out)
`
while in ConditionBlock:
`    def forward(self, x, c):
        shift_msa, scale_msa, gate_msa, shift_mlp, scale_mlp, gate_mlp = (
            self.adaLN_modulation(c).chunk(6, dim=-1)
        )
        x = x + gate_msa * self.attn(modulate(self.norm1(x), shift_msa, scale_msa))
        x = x + gate_mlp * self.mlp(modulate(self.norm2(x), shift_mlp, scale_mlp))
        return x
`
I am confused because to my knowledge, it may infect the condition modulation by AdaLN when using a duplicated LayerNorm right after AdaLN. Is it a special design? For I am just a new hand in DL, if I have any misunderstanding, feel free to reach out! ：)

**Comment by lucas-maes** [comment]
hi! Indeed, we are aware of this! We already discussed that in a previous issue! From the best of my understanding, it didn't impact performance!
